Sources/FoundationEssentials/Subprocess/Subprocess+Configuration.swift

```diff
@@ -0,0 +1,615 @@
+                    }
+                    // Ignore ESRCH (no such process)
+                    if posixError.code != .ESRCH {
+                        killError = error
+                    }
+                }
+            }
+
+            if let inputError = inputError {
+                throw inputError
+            }
+
+            if let outputError = outputError {
+                throw outputError
+            }
+
+            if let errorError = errorError {
+                throw errorError
+            }
+
+            if let killError = killError {
+                throw killError
+            }
+        }
+
+        /// Close each input individually, and throw the first error if there's multiple errors thrown
+        @Sendable
+        internal func cleanupAll(
+            input: ExecutionInput,
+            output: ExecutionOutput,
+            error: ExecutionOutput
+        ) throws {
+            var inputError: Error?
+            var outputError: Error?
+            var errorError: Error?
+
+            do {
+                try input.closeAll()
+            } catch {
+                inputError = error
+            }
+
+            do {
+                try output.closeAll()
+            } catch {
+                outputError = error
+            }
+
+            do {
+                try error.closeAll()
+            } catch {
+                errorError = error
+            }
+
+            if let inputError = inputError {
+                throw inputError
+            }
+            if let outputError = outputError {
+                throw outputError
+            }
+            if let errorError = errorError {
+                throw errorError
+            }
+        }
+
+        internal func run<R>(
+            output: RedirectedOutputMethod,
+            error: RedirectedOutputMethod,
+            _ body: @Sendable @escaping (Subprocess, StandardInputWriter) async throws -> R
+        ) async throws -> Result<R> {
+            let (readFd, writeFd) = try FileDescriptor.pipe()
+            let executionInput: ExecutionInput = .customWrite(readFd, writeFd)
+            let executionOutput: ExecutionOutput = try output.createExecutionOutput()
+            let executionError: ExecutionOutput = try error.createExecutionOutput()
+            let process: Subprocess = try self.spawn(
+                withInput: executionInput,
+                output: executionOutput,
+                error: executionError)
+            // After spawn, cleanup child side fds
+            try self.cleanup(
+                process: process,
+                childSide: true,
+                parentSide: false,
+                attemptToTerminateSubProcess: false
+            )
+            return try await withTaskCancellationHandler {
+                return try await withThrowingTaskGroup(of: RunState<R>.self) { group in
+                    group.addTask {
+                        let status = monitorProcessTermination(
```

Comment:
Could you elaborate on this? Are you saying all blocking methods need to be in a separate executor? That would make Swift's concurrency system really hard to use...

Reply:
I know we discussed an alternative solution for this offline and we should use kqueue,epoll,io_uring to monitor to process instead. However, to answer your general question for anyone that's reading along: Yes you can never make a blocking sys call in neither a sync nor async method really since those might be called from one of the Concurrency threads which will get blocked. Any blocking call must run on its own thread and the canonical spelling starting is going to be `withTaskExecutor` once that feature has landed in a released Swift version.